aten/src/ATen/cuda/CUDASparseBlas.h

```diff
@@ -315,6 +315,33 @@ template <>
 void bsrsm2_solve<c10::complex<double>>(
     CUSPARSE_BSRSM2_SOLVE_ARGTYPES(c10::complex<double>));
 
+#define CUSOLVER_LINEAR_SOLVE_ARGTYPES(scalar_t)                                 \
+    cusolverSpHandle_t &handle, int n, int nnzA, const cusparseMatDescr_t descrA, \
+    const scalar_t *csrValA, const int *csrRowPtrA, const int *csrColIndA,       \
+    const scalar_t *b, float tol, int reorder, scalar_t *x, int *singularity
```

Comment:
It's not correct to use `float tol`. I suggest using the same signature as in the library, so `cusolverSpHandle_t &handle` -> `cusolverSpHandle_t handle`.
```suggestion
#define CUSOLVER_LINEAR_SOLVE_ARGTYPES(scalar_t, value_t)                                 \
    cusolverSpHandle_t handle, int n, int nnzA, const cusparseMatDescr_t descrA, \
    const scalar_t *csrValA, const int *csrRowPtrA, const int *csrColIndA,       \
    const scalar_t *b, value_t tol, int reorder, scalar_t *x, int *singularity
```

Reply:
My bad, just noticed that `tol` can be `float`/`double` based on the input dtype.